Update .NET Core index page (#16680)
dotnet/docs: docs/core/about.md

> ---

Reviewer: update ms.date on line 4

# About .NET Core

> .NET Core is/has:

Reviewer: The original sentence was better for loc.

- **Cross-platform:** Runs on Windows, macOS, and Linux [operating systems](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md).
- **Open source:** The .NET Core platform is [open source](https://github.com/dotnet/core), using MIT and Apache 2 licenses. .NET Core is a [.NET Foundation](https://dotnetfoundation.org/) project.

> - **Modern:** It implements modern paradigms like asyncronous programming, no-copy patterns using structs, and resource governance for containers.

Reviewer: ```suggestion
- **Modern:** It implements modern paradigms like asynchronous programming, no-copy patterns using structs, and resource governance for containers.
```

> - **High performance:** Uses [hardware intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/) and [constant investment](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-core-3-0/) to improve performance with every release.

Reviewer 2: After reading the "hardware intrinsics" words, the "constant investment" reads for a moment as another feature or pattern to write a high performance code in .NET Core.

Author: I agree. I didn't like that when I wrote it. Any thoughts on something better?

Reviewer 2: As far as I understand, this paragraph says two things:

- .NET Core gives developers the tools to write high-performance code (that is, hardware intrinsics, `Span<T>`, pipelines, etc)
- the development team continuously improves the performance of .NET Core itself

However, I don't have good thoughts now on how to say that in one paragraph.

Author: How is this? https://github.com/dotnet/docs/pull/16680/files#r367078380

> - **High performance:** Uses [hardware intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/) and [constant investment](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-core-3-0/) to improve performance with every release.

Author: ```suggestion
- **High performance:**  [Improves performance] (https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-core-3-0/) with new features like [hardware intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/), [tiered compilation](https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/tiered-compilation.md), and [Span<T>](https://docs.microsoft.com/archive/msdn-magazine/2018/january/csharp-all-about-span-exploring-a-new-net-mainstay).
```

Reviewer 2: Again, listing is not very homogeneous. While hardware intrinsics and `Span<T>` are things a developer can use, tiered compilation is the internal specifics of the runtime. Also, "new features" become old fast. Is `Span<T>` still new? Or the comparison ("improves", "new") is implicitly done with .NET Framework?

Author: Can remove "new". Agree.

All 3 of those things are performance-oriented investments we made that are intended to make your app faster. Hardware intrinsics and span are equally runtime features and they are also used extensively in the BCL so that they benefit you by default w/o you using them. So, I see these all as "same enough". Fair?

Reviewer 2: > So, I see these all as "same enough". Fair?

I agree and admit I might have been too nit-picky.

By the way, the docs have already page about `Span<T>` and related types:
https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/

Should it be linked instead of the magazine article?

Author: Good suggestion. good now?

Reviewer 2: Yes, with two nits:
- Please remove the white space after "[high performance]". I'm afraid it breaks the link formatting now.
- Use the relative link: `../standard/memory-and-spans/index.md`. That way, the links are checked at the build.

Author: OK. Did that (I think).

> - **Performance:**  Delivers [high performance](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-core-3-0/) with features like [hardware intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/), [tiered compilation](https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/tiered-compilation.md), and [Span<T>](../standard/memory-and-spans/index.md).

Reviewer: ```suggestion
- **Performance:**  Delivers [high performance](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-core-3-0/) with features like [hardware intrinsics](https://devblogs.microsoft.com/dotnet/hardware-intrinsics-in-net-core/), [tiered compilation](https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/tiered-compilation.md), and [Span\<T>](../standard/memory-and-spans/index.md).
```

> - **Consistent across environments:** Runs your code with the same behavior on multiple OSes and architectures, including x64, x86, and ARM.

Reviewer: The acronym for operating systems is actually OSs which can be usually confused with OSS. I'd just spell out.
https://itknowledgeexchange.techtarget.com/writing-for-business/the-plural-of-os/

```suggestion
- **Consistent across environments:** Runs your code with the same behavior on multiple operating systems and architectures, including x64, x86, and ARM.
```

- **Command-line tools:**  Includes easy-to-use command-line tools that can be used for local development and for continuous-integration.
- **Flexible deployment:** You can include .NET Core in your app or install it side-by-side (user-wide or system-wide installations). Can be used with [Docker containers](docker/introduction.md).

## Languages

C#, Visual Basic, and F# languages can be used to write applications and libraries for .NET Core. These languages can be used in your favorite text editor or Integrated Development Environment (IDE), including:

- [Visual Studio](https://visualstudio.microsoft.com/vs/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)
- [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)

Editor integration is provided, in part, by the contributors of the [OmniSharp](https://www.omnisharp.net/) and [Ionide](http://ionide.io) projects.

## APIs

.NET Core exposes frameworks for building any kind of app:

> * [Cloud apps, with ASP.NET Core](/aspnet/core/)

Reviewer: I'd remove the commas here 

```suggestion
* Cloud apps with [ASP.NET Core](/aspnet/core/)
```

> * [Mobile apps, with Xamarin](/xamarin)

Reviewer: ```suggestion
* Mobile apps with [Xamarin](/xamarin)
```

> * [IoT apps, with System.Device.GPIO](/archive/msdn-magazine/2019/august/net-core-cross-platform-iot-programming-with-net-core-3-0)

Reviewer: for links in the archive, I'd use the full link. we only use the cross repo for things that would be in the VS offline books

```suggestion
* IoT apps with [System.Device.GPIO](https://docs.microsoft.com/archive/msdn-magazine/2019/august/net-core-cross-platform-iot-programming-with-net-core-3-0)
```

> * [Windows client apps, with WPF and Windows Forms](https://docs.microsoft.com/dotnet/desktop-wpf/overview/index)

Reviewer: ```suggestion
* Windows client apps with [WPF](../desktop-wpf/overview/index.md) and Windows Forms.
```

> * [Machine learning, ML.NET](https://docs.microsoft.com/dotnet/machine-learning/).

Reviewer: ```suggestion
* Machine learning with [ML.NET](../machine-learning/index.md).
```

Many APIs are included that satisfy common needs, such as:

- Primitive types, such as <xref:System.Boolean?displayProperty=nameWithType> and <xref:System.Int32?displayProperty=nameWithType>.
- Collections, such as <xref:System.Collections.Generic.List%601?displayProperty=nameWithType> and <xref:System.Collections.Generic.Dictionary%602?displayProperty=nameWithType>.
- Utility types, such as <xref:System.Net.Http.HttpClient?displayProperty=nameWithType>, and <xref:System.IO.FileStream?displayProperty=nameWithType>.

> - Data types, such as <xref:System.Data.DataSet?displayProperty=nameWithType>, and [DbSet](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore/).

Reviewer: should we link to the API here too instead of the package?

> - High-performance types, such as <xref:System.Span%601?displayProperty=nameWithType>,  <xref:System.Numerics.Vector?displayProperty=nameWithType> and [Pipelines](../standard/io/pipelines.md).

Reviewer: ```suggestion
- High-performance types, such as <xref:System.Span%601?displayProperty=nameWithType>, <xref:System.Numerics.Vector?displayProperty=nameWithType> and [Pipelines](../standard/io/pipelines.md).
```

- The [.NET Core CLI tools](https://github.com/dotnet/sdk) and language compilers ([Roslyn](https://github.com/dotnet/roslyn) and [F#](https://github.com/microsoft/visualfsharp)) that enable the .NET Core developer experience.
- The [dotnet tool](https://github.com/dotnet/core-setup), which is used to launch .NET Core apps and CLI tools. It selects the runtime and hosts the runtime, provides an assembly loading policy, and launches apps and tools.

### Open source

[.NET Core](https://github.com/dotnet/core) is open source ([MIT license](https://github.com/dotnet/core/blob/master/LICENSE.TXT)) and was contributed to the [.NET Foundation](https://dotnetfoundation.org) by Microsoft in 2014. It's now one of the most active .NET Foundation projects. It can be used by individuals and companies, including for personal, academic, or commercial purposes. Multiple companies use .NET Core as part of apps, tools, new platforms, and hosting services. Some of these companies make significant contributions to .NET Core on GitHub and provide guidance on the product direction as part of the [.NET Foundation Technical Steering Group](https://dotnetfoundation.org/blog/tsg-welcome).

## Support

> .NET Core is [supported by Microsoft](https://dotnet.microsoft.com/platform/support/policy), on Windows, macOS, and Linux. It's updated for security and quality regularly (second tuesday of each month).

Reviewer: ```suggestion
.NET Core is [supported by Microsoft](https://dotnet.microsoft.com/platform/support/policy), on Windows, macOS, and Linux. It's updated for security and quality regularly (second Tuesday of each month).
```
